lib/runtime-core/src/backend.rs

```diff
@@ -148,6 +155,30 @@ pub struct CompilerConfig {
     pub generate_debug_info: bool,
 }
 
+impl Default for CompilerConfig {
+    fn default() -> Self {
+        Self {
+            symbol_map: Default::default(),
```

Comment:
Couldn't you use `..Default::default()` for these?

Reply:
That would infinite loop I believe!  My understanding of what `..Default::default()` does is that it calls default on the type you're constructing and then overrides the fields you specified.  Because we're defining default, I don't believe we can use that syntax here.

Reply:
Yeah, just verified this:

```
warning: function cannot return without recursing
   --> lib/runtime-core/src/backend.rs:159:5
    |
159 |     fn default() -> Self {
    |     ^^^^^^^^^^^^^^^^^^^^ cannot return without recursing
...
178 |             ..Default::default()
    |               ------------------ recursive call site
    |
    = note: `#[warn(unconditional_recursion)]` on by default
    = help: a `loop` may express intention better if this is on purpose
```